pkg/controller/cloud/node_controller.go

```diff
@@ -284,6 +284,16 @@ func (cnc *CloudNodeController) AddCloudNode(obj interface{}) {
 			return err
 		}
 
+		providerID, err := cloudprovider.GetInstanceProviderID(cnc.cloud, types.NodeName(curNode.Name))
+		if err == nil {
+			curNode.Spec.ProviderID = providerID
+		} else {
+			// we should attempt to set providerID on curNode, but
+			// we can continue if we fail since we will attempt to set
+			// node addresses given the node name in getNodeAddressesByProviderIDOrName
+			glog.Errorf("%v", err)
```

Comment:
This is changing logic from the kubelet. If the kubelet encounters an error getting the ProviderID it won't continue fetching subsequent information from the cloud provider. https://github.com/kubernetes/kubernetes/blob/02b520f0a40be2056d91fc0661c2b4fdb2694c30/pkg/kubelet/kubelet_node_status.go#L295